orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,161 @@
-from nltk.stem.porter import PorterStemmer
+from nltk.corpus import stopwords
 from nltk.stem import WordNetLemmatizer
-from sklearn.feature_extraction.text import CountVectorizer
+from nltk.tokenize import word_tokenize
+from nltk.stem.porter import PorterStemmer
+from nltk.stem.snowball import SnowballStemmer
+
+from orangecontrib.text.corpus import Corpus
+
+import Orange
 
 
 class Preprocessor:
     """
-        Holds pre-processing flags and other information, about stop word
-        removal, lowercasing, text morphing etc.(the options are set via
-        the Preprocess widget).
+        A pre-processing class for the Orange text mining extension. An instance of the class
+        is capable of tokenizing, lowercasing, stemming and lemmatizing the input. Returns a list
+        of preprocessed tokens, sorted by order of appearance in text.
     """
-    def __init__(self, incl_punct=False, lowercase=True, stop_words='english', trans=None, min_df=1):
+    def __init__(self, tokenizer="word_tokenizer", lowercase=True, stop_words=None, transformation=None):
```

Comment:
`word_tokenizer` seem quite long argument to write and remember. Couldn't we use just a flag insted? For example, `use_twitter_tokenize=False`?